std/digest/xxh.d

```diff
@@ -0,0 +1,3352 @@
+    size_t nbStripesSoFar;
+    /*!< Number or stripes processed. */
+    XXH64_hash_t totalLen;
+    /*!< Total length hashed. 64-bit even on 32-bit targets. */
+    size_t nbStripesPerBlock;
+    /*!< Number of stripes per block. */
+    size_t secretLimit;
+    /*!< Size of @ref customSecret or @ref extSecret */
+    XXH64_hash_t seed;
+    /*!< Seed for _withSeed variants. Must be zero otherwise, @see XXH3_INITSTATE() */
+    XXH64_hash_t reserved64;
+    /*!< Reserved field. */
+    const(ubyte)* extSecret;
+    /*!< Reference to an external secret for the _withSecret variants, null
+        *   for other variants. */
+    /* note: there may be some padding at the end due to alignment on 64 bytes */
+} /* typedef'd to XXH3_state_t */
+
+enum XXH_errorcode
+{
+    XXH_OK = 0, /*!< OK */
+    XXH_ERROR /*!< Error */
+}
+
+/*!
+ * @internal
+ * @brief Structure for XXH32 streaming API.
+ *
+ * @note This is only defined when @ref XXH_STATIC_LINKING_ONLY,
+ * @ref XXH_INLINE_ALL, or @ref XXH_IMPLEMENTATION is defined. Otherwise it is
+ * an opaque type. This allows fields to safely be changed.
+ *
+ * Typedef'd to @ref XXH32_state_t.
+ * Do not access the members of this struct directly.
+ * @see XXH64_state_s, XXH3_state_s
+ */
+struct XXH32_state_t
+{
+    XXH32_hash_t total_len_32; /*!< Total length hashed, modulo 2^32 */
+    XXH32_hash_t large_len; /*!< Whether the hash is >= 16 (handles @ref total_len_32 overflow) */
+    XXH32_hash_t[4] v; /*!< Accumulator lanes */
+    XXH32_hash_t[4] mem32; /*!< Internal buffer for partial reads. Treated as unsigned char[16]. */
+    XXH32_hash_t memsize; /*!< Amount of data in @ref mem32 */
+    XXH32_hash_t reserved; /*!< Reserved field. Do not read nor write to it. */
+} /* typedef'd to XXH32_state_t */
+
+/*!
+ * @internal
+ * @brief Structure for XXH64 streaming API.
+ *
+ * @note This is only defined when @ref XXH_STATIC_LINKING_ONLY,
+ * @ref XXH_INLINE_ALL, or @ref XXH_IMPLEMENTATION is defined. Otherwise it is
+ * an opaque type. This allows fields to safely be changed.
+ *
+ * Typedef'd to @ref XXH64_state_t.
+ * Do not access the members of this struct directly.
+ * @see XXH32_state_s, XXH3_state_s
+ */
+struct XXH64_state_t
+{
+    XXH64_hash_t total_len; /*!< Total length hashed. This is always 64-bit. */
+    XXH64_hash_t[4] v; /*!< Accumulator lanes */
+    XXH64_hash_t[4] mem64; /*!< Internal buffer for partial reads. Treated as unsigned char[32]. */
+    XXH32_hash_t memsize; /*!< Amount of data in @ref mem64 */
+    XXH32_hash_t reserved32; /*!< Reserved field, needed for padding anyways*/
+    XXH64_hash_t reserved64; /*!< Reserved field. Do not read or write to it. */
+} /* typedef'd to XXH64_state_t */
+
+struct XXH32_canonical_t
+{
+    ubyte[4] digest; /*!< Hash bytes, big endian */
+}
+
+/** A 32-bit byteswap.
+ *
+ * Param: x = The 32-bit integer to byteswap.
+ * Return: x, byteswapped.
+ */
+private xxh_u32 XXH_swap32(xxh_u32 x) @trusted pure nothrow @nogc
```

Comment:
There is already a `bswap` intrinsic. https://dlang.org/phobos/core_bitop.html#.bswap

Reply:
Changed.